stdlib/public/core/Array.swift

```diff
@@ -1599,6 +1595,132 @@ extension Array {
     )
   }
 
+  @_alwaysEmitIntoClient
+  @available(SwiftCompatibilitySpan 5.0, *)
+  internal init<E: Error>(
+    _uninitializedCapacity capacity: Int,
+    initializingWith initializer: (
+      _ span: inout OutputSpan<Element>
+    ) throws(E) -> Void
+  ) throws(E) {
+    var firstElementAddress: UnsafeMutablePointer<Element>
+    unsafe (self, firstElementAddress) =
+      unsafe Array._allocateUninitialized(capacity)
+    let buffer = unsafe UnsafeMutableBufferPointer<Element>(
+      start: firstElementAddress, count: capacity
+    )
+
+    var span = unsafe OutputSpan<Element>(buffer: buffer, initializedCount: 0)
+
+    // The following should be a `defer` block.
+    // Workaround for https://github.com/swiftlang/swift/issues/76388
+    func finalize(_ span: consuming OutputSpan<Element>) {
+      // Update self.count even if `initializer` throws an error.
+      self._buffer.mutableCount = unsafe span.finalize(for: buffer)
+      _endMutation()
+    }
+
+    // If `defer` worked above, then this wouldn't be a do/catch block.
+    // Workaround for https://github.com/swiftlang/swift/issues/76388
+    do {
+      try initializer(&span)
+      finalize(span)
+    }
+    catch {
+      finalize(span)
+      throw error
+    }
+  }
+
+  /// Creates an array with the specified capacity, then calls the given
+  /// closure with an OutputSpan covering the array's uninitialized memory.
```

Comment:
Two changes here (that should be applied to the `append` method and the two symbols in `ContiguousArray`):
- Apple docs style uses "and then", since "then" isn't a conjunction (even though what you've written is widely accepted)
- Type names should always be written as plain English ("output span") or in code voice, and code voice is prohibited in abstracts.

```suggestion
  /// Creates an array with the specified capacity, and then calls the given
  /// closure with an output span covering the array's uninitialized memory.
```

Reply:
Looks like we should update the docs for point 1 in the unsafe inits, as well!